modules/data.land/R/soil_utils.R

```diff
@@ -0,0 +1,315 @@
+  
+  #---------------------------------------------------------------------------------------#
+  #     Here there is not much we can do other than explore where in the triangle space   #
+  # we are.                                                                               #
+  #---------------------------------------------------------------------------------------#
+  
+  if (any(silt > 100.) | any(silt < 0.) | any(sand > 100.) | 
+      any(sand < 0.) | any(clay > 100.) | any(clay < 0.) ) {
+    print("---------------------------------------------------")
+    print(" At least one of your percentages is screwy...")
+    print(paste("SAND <- ",sprintf("%.2f",sand),"%",sep=""))
+    print(paste("CLAY <- ",sprintf("%.2f",clay),"%",sep=""))
+    print(paste("SILT <- ",sprintf("%.2f",silt),"%",sep=""))
+    print("---------------------------------------------------")
+    stop ("This soil doesn''t fit into any category...")
+    
+  }
+  nlayer = max(length(silt),length(clay),length(sand))
+  mysoil = NA
+  for(z in seq_len(nlayer)){
+    if(sand[z] > 85.0 + 0.5 * clay[z]) {
+      mysoil[z] <-  1 #----- Sand. ------------------------------------------------------------#
+    }else if(sand[z] > 70.0 + clay[z]) {
+      mysoil[z] <-  2 #----- Loamy sand. ------------------------------------------------------#
+    }else if((clay[z] <= 20.0 & sand[z] > 52.5) | (clay[z] <= 7.5 & silt[z] <= 50.0)) {
+      mysoil[z] <-  3 #----- Sandy loam. ------------------------------------------------------#
+    }else if((clay[z] <= 27.5 & silt[z] > 50.0 & silt[z] <= 80.0) | (silt[z] >  80.0 & clay[z] > 12.5)) {
+      mysoil[z] <-  4 #----- Silt loam. -------------------------------------------------------#
+    }else if(clay[z] > 7.5 & clay[z] <= 27.5 & silt[z] > 27.5 & silt[z] <= 50.0 & sand[z] <= 52.5) {
+      mysoil[z] <-  5 #----- Loam. ------------------------------------------------------------#
+    }else if(clay[z] > 20.0 & clay[z] <= 35.0 & silt[z] <= 27.5 & sand[z] > 45.0) {
+      mysoil[z] <-  6 #----- Sandy clay loam. -------------------------------------------------#
+    }else if(clay[z] > 27.5 & clay[z] <= 40.0 & sand[z] <= 20.0) {
+      mysoil[z] <-  7 #----- Silty clay loam. -------------------------------------------------#
+    }else if(clay[z] > 27.5 & clay[z] <= 40.0 & sand[z] > 20.0 & sand[z] <= 45.0) {
+      mysoil[z] <-  8 #----- Clayey loam. -----------------------------------------------------#
+    }else if(clay[z] > 35.0 & sand[z] > 45.0) {
+      mysoil[z] <-  9 #----- Sandy clay. ------------------------------------------------------#
+    }else if(clay[z] > 40.0 & silt[z] > 40.0) {
+      mysoil[z] <- 10 #----- Silty clay. ------------------------------------------------------#
+    }else if(clay[z] <= 70.0 & sand[z] <= 30.0 & silt[z] <= 30.0) {
+      mysoil[z] <- 11 #----- Clay. ------------------------------------------------------------#
+    }else if( silt[z] > 80.0 & clay[z] <= 12.5) {
```

Comment:
Not sure if my last comment got added, but basically this function uses texture to get info, but doesn't have a way to use OC information to figure out peaty soils. Something to consider for future change, working in wetlands.